docs/weave/weave-api-spec/03-querying.md

```diff
@@ -0,0 +1,191 @@
+---
+id: weave-query-spec
+title: Querying Weave
+sidebar_label: Querying Weave
+---
+
+In this section querying weave for data will be explained. Before diving into the details, Weave internals such as __buckets__ and __paths__ must be explained.
+
+## Buckets
+
+Buckets are the structures that enable accessing and writing to __Key-Value__ database in an organized and controlled manner. Buckets stores the data as well as the indexes, secondary indexes that refer to the data.
+
+### Accessing buckets
+
+[//]: # (TODO give reference to weave/tendermint or abci documentation)
+As mentioned in the previous sections, Weave uses tendermint as consensus engine thus queries are made to data store via `abci_queries`. Therefore when you make a query actually you do the call to tendermint's abci protocol. For more info about underlyings refer to [tendermint/abciquery](https://tendermint.com/rpc/#abciquery).
+
+Via running the JSON-RPC/HTTP call below, __hugnet__ testnet could be queried so you can see an example response.
+
+``` bash
+curl -X POST -d '{ "json-rpc": 2.0, "id": "foobar321", "method": "abci_query", "params": { "path": "/tokens", "data": "" } }' https://bns.hugnet.iov.one/
+```
+
+### Bucket paths
+
+Every bucket registered under Weave is accessed using __paths__. On the example curl command above given path parameter says the bucket wanted to be accessed is the tokens bucket.
```

Comment:
I think this is a bit convoluted paragraph. How about

>  A bucket data can be queried using that bucket unique path to identify it.
> The above `curl` command is reading username `Token` entity using `/tokens` path.

Bucket does not have to be registered. We usually expose all buckets we create, but this is not a requirement.